writer.go

```diff
@@ -719,20 +719,34 @@ func (w *writer) writeWithRetries(conn *Conn, batch []Message, resch [](chan<- e
 		w.stats.retries.observe(1)
 		time.Sleep(backoff(attempt+1, 100*time.Millisecond, 1*time.Second))
 	}
+
+	if err != nil {
+		for i, res := range resch {
+			res <- &writerError{msg: batch[i], err: err}
+		}
+	} else {
+		for _, res := range resch {
+			res <- nil
+		}
+		for _, m := range batch {
+			w.stats.messages.observe(1)
```

Comment:
Can we take this out of the loop? It seems like it could be `w.stats.messages.observe(int64(len(batch)))`